### PR TITLE
Add ambient mob sounds

### LINKED ISSUE #3443: Mobs don't make sounds when they move or do periodicals
A zombie groaning occasionally, a sheep baaa'ing.

Taken from #1109

### PR DESCRIPTION
**Pull request by mathiascode**
Nothing too fancy, but does the job. Partially fixes. #3443.

### CODE REVIEW

File: src/Mobs/Monster.cpp [hunk 1]
```diff
@@ -384,6 +385,24 @@ void cMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 	BroadcastMovementUpdate();
 
+	// Ambient mob sounds
+	if (!m_SoundAmbient.empty())
+	{
+		--m_AmbientSoundTimer;
```
Comment (bearbin) [inline src/Mobs/Monster.cpp]: You could combine the decrement into the if condition here.

File: src/Mobs/Monster.cpp [hunk 2]
```diff
@@ -384,6 +385,24 @@ void cMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
...
+
+		if (m_AmbientSoundTimer <= 0)
+		{
+			auto & Random = GetRandomProvider();
+			auto ShouldPlaySound = Random.RandBool();
+			if (ShouldPlaySound)
+			{
+				auto SoundPitchMultiplier = 1.0f + (Random.RandReal(1.0f) - Random.RandReal(1.0f)) * 0.2f;
+				m_World->BroadcastSoundEffect(m_SoundAmbient, GetPosition(), 1.0f, SoundPitchMultiplier * 1.0f);
```
Comment (bearbin) [inline src/Mobs/Monster.cpp]: A multiplication by one shouldn't do anything to SoundPitchMultiplier?

File: src/Mobs/Squid.cpp [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 
 cSquid::cSquid(void) :
-	super("Squid", mtSquid, "entity.squid.hurt", "entity.squid.death", 0.95, 0.95)
+	super("Squid", mtSquid, "entity.squid.hurt", "entity.squid.death", "", 0.95, 0.95)
```
Comment (bearbin) [inline src/Mobs/Squid.cpp]: Do squids not have an ambient sound?

File: src/Mobs/Monster.h [hunk 4]
```diff
@@ -287,6 +288,8 @@ class cMonster :
 	bool m_BurnsInDaylight;
 	double m_RelativeWalkSpeed;
 
+	int m_AmbientSoundTimer = GetRandomProvider().RandInt(0, 100);
```
Comment (bearbin) [inline src/Mobs/Monster.h]: This should really be in the constructor.

**Comment by Greaper88** [comment]
"what does the squid say?!"

File: src/Mobs/Monster.cpp (hunk 2 above)
Comment (mathiascode) [reply src/Mobs/Monster.cpp]: Ah right, leftover from my explosion changes. Will fix.

File: src/Mobs/Squid.cpp (hunk 3 above)
Comment (mathiascode) [reply src/Mobs/Squid.cpp]: Seems like it was introduced in 1.13. I'll add the sound name.

**Pull request merged**